### PR TITLE
DISCARD ME

### PR DESCRIPTION
**Pull request by pavel-savinov**

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 1]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
 		if (this._step === 1) {
 			this._step = 2;
 		} else {
-			addStructuredContent(
-				this.refs.modal.refs.mapContentForm.getSerializedFields(),
-				this._ddmStructure.id,
-				this._title
-			).then(() => {});
+			this.store.dispatch(enableSavingChangesStatusAction()).done(() => {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: This is not necessary, you can just dispatch before

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 2]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+				addStructuredContent(
+					this.refs.modal.refs.mapContentForm.getSerializedFields(),
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: We shouldn't be calling methods from other components

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 3]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+					this._ddmStructure.id,
+					this._title
+				).then(response => {
+					if (response.errorMessage) {
+						this._errorMessage = response.errorMessage;
+					} else if (!response.classNameId || !response.classPK) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: We may add this "unexpected" error message inside addStructuredContent and throw an error

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 4]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+						this._errorMessage = Liferay.Language.get(
+							'an-unexpected-error-occurred'
+						);
+					} else {
+						const promises = [];
+						const updateFragmentEntryLinkIds = [];
+
+						this._fields.forEach(field => {
+							if (field.fragmentEntryLinkId) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: Better use filter

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 5]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+								const currentValues = this.fragmentEntryLinks[
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: Do not mutate objects :fearful:

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 6]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+								];
+
+								currentValues[field.editableId].classNameId =
+									response.classNameId;
+								currentValues[field.editableId].classPK =
+									response.classPK;
+								currentValues[field.editableId].fieldId =
+									field.key;
+
+								if (
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: Array.includes()

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 7]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+									) === -1
+								) {
+									updateFragmentEntryLinkIds.push(
+										field.fragmentEntryLinkId
+									);
+								}
+							}
+						});
+
+						updateFragmentEntryLinkIds.forEach(
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: Chain with previous loop (when moving errors to .catch)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 8]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+							(fragmentEntryLinkId, index) => {
+								promises.push(
+									updateEditableValues(
+										fragmentEntryLinkId,
+										this.fragmentEntryLinks[
+											fragmentEntryLinkId
+										].editableValues,
+										index ===
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: ¿?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 9]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+											updateFragmentEntryLinkIds.length -
+												1
+									)
+								);
+							}
+						);
+
+						Promise.all(promises).then(() => {
+							this.store
+								.done(() => {})
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: Unnecesary call

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js [hunk 10]
```diff
@@ -72,11 +82,80 @@ class CreateContentDialog extends Component {
...
+								.dispatch(updateMappedContentsAction())
+								.dispatch(
+									Object.assign({}, response, {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentDialog.es.js]: Use object spread now that we can :)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentForm.es.js [hunk 11]
```diff
@@ -43,7 +43,10 @@ class CreateContentForm extends PortletBase {
 	 * @param {Event} event
 	 */
 	_handleFormChange(event) {
-		const form = event.delegateTarget;
+		const form =
+			event.delegateTarget.tagName === 'FORM'
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentForm.es.js]: Why is this necessary?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentForm.soy [hunk 12]
```diff
@@ -39,6 +39,7 @@
 
 				<input
 					class="form-control"
+					data-onkeyup="{$_handleFormChange}"
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/CreateContentForm.soy]: If you need to listen to form changes just add this listener to form

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js [hunk 13]
```diff
@@ -81,7 +80,7 @@ class MapContentForm extends PortletBase {
 						fields.push(field);
 					});
 
-					this._fields = fields;
+					this.fields = fields;
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js]: Dot not modify non-internal properties

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js [hunk 14]
```diff
@@ -52,9 +53,7 @@ class MapContentForm extends PortletBase {
 	 * @review
 	 */
 	syncDdmStructure() {
-		if (this.ddmStructure) {
-			this._fields = null;
-
+		if (this.ddmStructure && !this.fields) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js]: If ddmStructure changes we are not reloading fields never

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js [hunk 15]
```diff
@@ -133,6 +132,7 @@ class MapContentForm extends PortletBase {
 							languageId
 					  ]
 					: editableValues[editableId].defaultValue.trim();
+				selectedItem.displayValue = stripHtml(selectedItem.itemValue);
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js]: SF new line before
@wincent is this a bug?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js [hunk 16]
```diff
@@ -145,22 +145,30 @@ class MapContentForm extends PortletBase {
...
 
 		if (fieldKey === '-') {
-			this._fields = this._fields.map(field => {
-				if (field.editableId === editableId) {
-					delete field.editableId;
-				}
+			this.emit('fieldsChanged', {
+				fields: this.fields.map(field => {
+					if (field.editableId === editableId) {
+						delete field.editableId;
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js]: Do not mutate objects

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js [hunk 17]
```diff
@@ -145,22 +145,30 @@ class MapContentForm extends PortletBase {
...
 			});
 		} else {
-			this._fields = this._fields.map(field => {
-				if (field.key === fieldKey) {
-					field.editableId = editableId;
-				}
+			this.emit('fieldsChanged', {
+				fields: this.fields.map(field => {
+					if (field.key === fieldKey) {
+						field.editableId = editableId;
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js]: Do not mutate objects

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js [hunk 18]
```diff
@@ -145,22 +145,30 @@ class MapContentForm extends PortletBase {
 		const fieldKey =
 			targetElement.options[targetElement.selectedIndex].value;
 		const editableId = targetElement.dataset.itemEditableId;
+		const fragmentEntryLinkId =
+			targetElement.dataset.itemFragmentEntryLinkId;
 
 		if (fieldKey === '-') {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js]: These two cases look quite similar, maybe we could generalize this

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.soy [hunk 19]
```diff
@@ -13,55 +24,61 @@
...
 								data-item-fragment-entry-link-id="{$selectedItem.fragmentEntryLinkId}"
 								data-onchange="{$_handleFieldChange}"
+
+								{if $savingChanges}
+									disabled
+								{/if}
 							>
-								{if $_fields}
-									{foreach $field in $_fields}
+								{if $fields}
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.soy]: Too much nesting, we need  to move this to a different template

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.soy [hunk 20]
```diff
@@ -13,55 +24,61 @@
...
 
-			{if $selectedItems and length($selectedItems) > 0}
-				{foreach $selectedItem in $selectedItems}
-					<div class="row mb-3">
-						<div class="col">
+		<div class="d-flex">
+			<div class="pr-1 w-50">
+				<h3 class="sheet-subtitle">{msg desc=""}content-value{/msg}</h3>
+
+				{if $selectedItems and length($selectedItems) > 0}
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.soy]: `> 0` not needed, you can even just "foreach" if you are not adding any wrapper element

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/floating_toolbar/mapping/FloatingToolbarMappingPanel.es.js [hunk 21]
```diff
@@ -289,6 +292,15 @@ class FloatingToolbarMappingPanel extends PortletBase {
 		});
 	}
 
+	/**
+	 * Opens content creation dialog
+	 * @private
+	 * @review
+	 */
+	_handleCreateContentClick() {
+		openCreateContentDialog(this.store, () => {}, () => {}, () => {});
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/floating_toolbar/mapping/FloatingToolbarMappingPanel.es.js]: Is this production code?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/floating_toolbar/mapping/FloatingToolbarMappingPanel.soy [hunk 22]
```diff
@@ -30,52 +31,78 @@
...
-					{param store: $store /}
-				{/call}
+		{if $contentCreationEnabled}
+			<div class="form-group">
+				<button
+					class="btn btn-block btn-secondary btn-sm"
+					data-onclick="{$_handleCreateContentClick}"
+
+					{if $item.editableValues.mappedField or $item.editableValues.fieldId}
+					disabled
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/floating_toolbar/mapping/FloatingToolbarMappingPanel.soy]: SF indent

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/fragment_entry_link/FragmentEditableField.es.js [hunk 23]
```diff
@@ -257,12 +256,22 @@ class FragmentEditableField extends PortletBase {
 		const processor =
 			FragmentProcessors[this.type] || FragmentProcessors.fallback;
 
+		const buttons = processor.getFloatingToolbarButtons(
+			this.editableValues
+		);
+
+		if (this.selectedItems.length > 1) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/fragment_entry_link/FragmentEditableField.es.js]: `>1` not necessary

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/fragment_entry_link/FragmentEditableField.es.js [hunk 24]
```diff
@@ -257,12 +256,22 @@ class FragmentEditableField extends PortletBase {
...
+			buttons.forEach(button => {
+				if (button.id !== FLOATING_TOOLBAR_BUTTONS.map.id) {
+					button.cssClass +=
+						' disabled fragments-editor__floating-toolbar--disabled';
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/fragment_entry_link/FragmentEditableField.es.js]: Investigate if we can remove this pattern

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/utils/FragmentsEditorFetchUtils.es.js [hunk 25]
```diff
@@ -302,12 +302,17 @@ function setStore(store) {
  * @param {object} editableValues
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/utils/FragmentsEditorFetchUtils.es.js]: Update jsdocs to understand what "updateClassedModel" does :musical_note:
Moreover, this is a boolean trap

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/utils/FragmentsEditorTextUtils.es.js [hunk 26]
```diff
@@ -0,0 +1,32 @@
...
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/**
+ * Strips HTML from the text
+ * @param {string} text
+ * @return {string}
+ */
+const stripHtml = text => {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/utils/FragmentsEditorTextUtils.es.js]: Definitely not a friend of this, but if we really need, investigate if this already exist in portal and/or use HTMLElement trick:
```
const e = document.createElement('div');
e.innerHTML = html;
e.innerText; // voilá
```

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js (hunk 15 above)
Comment (wincent) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/content/MapContentForm.es.js]: No I don't think so. The indent is correct, and Prettier isn't going to add or remove blank lines. (It generally tries to respect what the author does as far as grouping lines together goes.)

**Comment by p2kmgcl** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/77809
